NOTES.ZOS

```diff
@@ -0,0 +1,73 @@
+
+ NOTES FOR THE Z/OS PLATFORM
+ ===========================
+
+ First of all note that OpenSSL is currently NOT supported on z/OS.
+
+ However, if you want to build OpenSSL on z/OS anyway, here are a few
+ hints on how to do that. Though these hints are intended to be helpful,
+ they might not work for you at all, depending on your build environment.
+
+ Building OpenSSL 1.1.1
+ ----------------------
+ (with xplink linkage, 64 bit, from EBCDIC sources)
+
+ 1. get a working EBCDIC perl version.
+
+ 2. gunzip the .tar file.
+
+ 3. use pax to extract from .tar. Example:
+
+      pax -ofrom=ISO8859-1,to=IBM-1047 -rvf openssl-1.1.1e.tar
+
+ 4. adjust the configuration files. Examples:
+
+ 4.1 10-main.conf
+
+     "OS390-Unix" => {
+         inherit_from     => [ "BASE_unix", asm("zos_asm") ],
+         cc               => "./tools/c99.sh",
+         cflags           => "-O -Wc,dll,XPLINK,exportall,hgpr,lp64 -Wa,'GOFF,SYSPARM(USE_XPLINK)' -qlongname -qlanglvl=extc99 -DB_ENDIAN -DCHARSET_EBCDIC -DNO_SYS_PARAM_H -D_ALL_SOURCE -D_OPEN_THREADS=2 -D_POSIX_SOURCE  -D_OPEN_MSGQ_EXT",
+         module_ldflags   => "-Wl,XPLINK,LP64",
+         shared_ldflags   => "-Wl,dll,XPLINK,LP64",
+         bn_ops           => "THIRTY_TWO_BIT RC4_CHAR",
+         thread_scheme    => "(unknown)",
+         perlasm_scheme   => "zOS64",
+     },
+
+ 4.2 00-base-templates.conf
+
+     # Copy and adjust the s390x_asm section by commenting out all asm
+     # or use a "no-asm" build right away.
```

Comment:
This comment makes no sense at all.  What `s390x_asm` section?

Reply:
Its was just meant to point out that all s390 specific (gnu-)asm wont compile on zos.